core/ui_events.js

```diff
@@ -113,5 +114,78 @@ Blockly.Events.Click.prototype.fromJson = function(json) {
   this.blockId = json['blockId'];
 };
 
+/**
+ * Class for a marker move event.
+ * @param {?Blockly.Block=} opt_block The affected block. Null if current node
+ *    is of type workspace. Undefined for a blank event.
+ * @param {boolean=} isCursor Whether this is a cursor event.
+ * @param {?Blockly.ASTNode=} opt_oldNode The old node the marker used to be on.
+ * @param {!Blockly.ASTNode=} opt_curNode The new node the marker is now on.
+ * @extends {Blockly.Events.Ui}
+ * @constructor
+ */
+Blockly.Events.MarkerMove = function(opt_block, isCursor, opt_oldNode,
+    opt_curNode) {
+  var workspaceId = opt_block ? opt_block.workspace.id : undefined;
+  if (opt_curNode && opt_curNode.getType() == Blockly.ASTNode.types.WORKSPACE) {
+    workspaceId =
+        (/** @type {!Blockly.Workspace} */ (opt_curNode.getLocation())).id;
+  }
+  Blockly.Events.MarkerMove.superClass_.constructor.call(this, workspaceId);
+  this.blockId = opt_block ? opt_block.id : null;
```

Comment:
Should this also have a comment?

Reply:
I wasn't sure if it's supposed to because the superclass declares it already with comment, but I just checked `Blockly.Events.BlockBase` and it just repeats the same comment, so I'll do that too.